### PR TITLE
RavenDB-6451 implementation

### PR DESCRIPTION
**Pull request by myarichuk**

### CODE REVIEW

File: test/FastTests/Issues/RavenDB-6451.cs [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+                    return true;
+            }
+            return false;
+        }
+
+        //slightly modified version from to http://www.michael-whelan.net/replacing-appdomain-in-dotnet-core/
+        public static IEnumerable<Assembly> GetRuntimeNonMicrosoftAssemblies()
+        {
+            var assemblies = new List<Assembly>();
+            var dependencies = DependencyContext.Default.RuntimeLibraries;
```
Comment (ppekrol) [inline test/FastTests/Issues/RavenDB-6451.cs:100]: Maybe you can use same technique as in here https://github.com/ravendb/ravendb/blob/v4.0/test/SlowTests/Tests/NoNonDisposableTests.cs
and avoid additional dependency to 'Microsoft.Extensions.DependencyModel' ?

File: test/FastTests/Issues/RavenDB-6451.cs [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+                           type.Namespace.StartsWith("Voron") ||
+                           type.Namespace.StartsWith("Sparrow"))
+                    select type).ToArray();
+
+            foreach (var t in exceptionTypes)
+            {
+                try
+                {
+                    if(t == typeof(DocumentConflictException))
+                        Debugger.Break();
```
Comment (ayende) [inline test/FastTests/Issues/RavenDB-6451.cs:40]: Why is this here?
